homeassistant/components/spotify/media_player.py

```diff
@@ -258,9 +257,7 @@ def source(self) -> str | None:
     @property
     def source_list(self) -> list[str] | None:
         """Return a list of source devices."""
-        if not self._devices:
-            return None
-        return [device["name"] for device in self._devices]
+        return [device["name"] for device in self.data.devices.data]
```

Comment:
Unless we register a listener directly or eg inherit from CoordinatorEntity here, the coordinator won't continue updating on an interval automatically.

Reply:
Correct, however, the Spotify entity will update every 30 seconds already by itself.

Reply:
The data will be stale since the coordinator won't update unless there are registered listeners.

Reply:
Oh right! 🤦 
Will adjust that